### PR TITLE
swupd: teach ParseManifestFile to fill in the Name field

### PR DESCRIPTION
**Pull request by cmarcelo**
When parsing a file, fill in the Name field based on the
filename, essentially taking whatever comes after "Manifest.". If
doesn't match, just leave Name empty like before.

Small test added so we can keep track of edge cases in the future.

Signed-off-by: Caio Marcelo de Oliveira Filho <caio.oliveira@intel.com>

### CODE REVIEW

File: swupd/manifest.go [hunk 1]
```diff
@@ -204,13 +204,23 @@ func ParseManifestFile(path string) (*Manifest, error) {
...
+	m.Name = getNameForManifestFile(path)
 	err = f.Close()
 	if err != nil {
 		return nil, err
 	}
 	return m, nil
 }
 
+func getNameForManifestFile(path string) string {
+	prefix := "Manifest."
```
[the remark this answers is not in the document]
Comment (cmarcelo) [reply swupd/manifest.go]: That would work, but I actually wanted a "split from the end", that would be immune to strange edge cases as a directory with a "matching" name, e.g. "/home/caio/Manifest.Directory/Manifest.Good".

[the remark this answers is not in the document]
Comment (cmarcelo) [reply swupd/manifest.go]: Thinking more, filepath.Base could solve this for me (most of it).
